recap/src/Imp/Eval.hs

```diff
@@ -11,20 +13,87 @@ import Imp.Lang
 -- Variable mapping; shadowing is allowed
 type VarMap = M.Map String Int
 
--- Possible errors reported during evaluation 
+-- Possible errors reported during evaluation
 data Error = UndefinedVar String | ParsingErr String | DivByZero deriving (Show)
 
 -- Evaluation monad
 type EvalM = StateT VarMap (ExceptT Error IO)
 
+getMap :: String -> EvalM Int
+getMap varName = do
+    varMap <- get
+    case M.lookup varName varMap of
+        Just value -> return value
+        Nothing    -> throwError (UndefinedVar varName)
```

Comment:
You can avoid explicit pattern matching on `Maybe` constructors by using the `maybe` function